### ISSUE TITLE
Discard Traefik Pilot notifications in the UI

### ISSUE THREAD

**Issue by ptruman** [traefik/traefik#7360]
Is there a way to disable the "Connect to Traefik Pilot" button in the UI?

**Comment by ldez** [comment]
Hello,

Can you elaborate on the explanation of your request? Why do you want to disable this button?

**Comment by ptruman** [comment]
Hi

Without being connected the button creates a notification/alert style icon,
which is not desirable if the user does not want/intend to connect to Pilot.

Also if it is the users preference not to connect to Pilot, the button is
superfluous.


By default, pi-hole also seems to block pilot.traefik.io and cause "An
unexpected error has occurred" to appear in place of the button, which is
also confusing (until diagnosed).

**Comment by ldez** [comment]
FYI Pilot is disabled by default, only the button on the UI is always here.

**Comment by ptruman** [comment]
It's the UI button I'm querying.
If you don't want to use Pilot, why have a permanent notification you can't clear down?  It's not a lovely UX/clean UI.

**Comment by SantoDE** [comment]
Hello folks,

thanks for your interest in the project. We've had discussions yesterday and decided to work on several improvements on that matter.

As we understand, more of an issue for you is the current notification rather than the connect button itself. For that purpose, we will work on a proper discard for that notification by utilization the browsers local storage. Also, we're thinking about improving the usage with ad-blocker like systems (e.g. pi-hole). This could be done by testing the connection from the dashboard to pilot. 

I hope, that answers your questions :-)

**Comment by ptruman** [comment]
Hi

It's a bit of both - a button someone may never use is just 'annoying' -
but the notification does tend to require/demand a reaction/click and feels
'pushy' :)

I'm trying 2.3.1 now to see what may be tweaked.

**Comment by SantoDE** [comment]
Thanks for the input!. `2.3.1` does not have the improvements included, yet. It will probably be in a bit later version.

**Comment by codyja** [comment]
I'd love to see a `pilot_enable: false` sort of option too. We're running in a network that doesn't have outbound access, and the UI load hangs and eventually show the following because its trying to connect outbound to https://pilot.traefik.io

Appreciate this project!

![image](https://user-images.githubusercontent.com/23107237/95598218-48637280-0a15-11eb-9821-cdbbbd68f102.png)

**Comment by binlab** [comment]
I'm also voting for this request. Would be nice to have the configuration for disabling annoying button from the dashboard. I hope for your understanding. Thanks.

**Comment by ChristianCiach** [comment]
We see the same issue as described by @codyja. Our products are used in secured infrastructures without access to the internet. And now we see this eyesore: 
![grafik](https://user-images.githubusercontent.com/19341644/96352950-064ac880-10c8-11eb-85a0-b1f2b34d83df.png)

**Comment by phewitt** [comment]
bump. would also like to see the ability to disable / remove this button.

**Comment by bramceulemans** [comment]
I would also like to be able to hide the button.

**Comment by m13t** [comment]
I'd also like to see this gone in environments where there's no direct internet connectivity. We have no use for it, so it would be nice to disable to keep things clean and concise.

**Comment by lucemans** [comment]
+1

**Comment by franz-josef-kaiser** [comment]
As it's close to half a year after the issue has been filed, did you find a community pleasing solution that considers the comments and the emoji-votes on this issues and it's comments yet, @SantoDE?

**Comment by ajgon** [comment]
Although it's not perfect, you can leverage CSP browser API to block pilot iframe completely:

```yaml
http:
  middlewares:
    blockIframe:
      headers:
        customResponseHeaders:
          Content-Security-Policy: "frame-src none;"
  routers:
    dashboard:
      entrypoints: websecure
        [... usual config ...]
      middlewares: blockIframe@file
```

It would force the browser to remove Pilot iframe completely, leaving only empty space. This would of course block all iframes on traefik page, but I didn't notice any other except this one. You can also fine-tune `frame-src` tag if necessary.

**Comment by ooshkamils515** [comment]
I really don't like this, this basically ends up being an advertisement for Pilot embedded in my infrastructure, and causes traefik instances to phone-home on each dashboard load:

![image](https://user-images.githubusercontent.com/25654424/111474814-d0efe880-8702-11eb-90e8-e6fea9ba4023.png)

This is phoning home metadata about the install with every dashboard load.  How is this acceptable?

**Comment by fhemberger** [comment]
I'm already looking into downgrading (to v2.2.x)/replacing Traefik because of this.
Connecting to pilot should be an opt-in decision for operators, in the meantime we are DNS blocking/firewalling outgoing requests to `pilot.traefik.io`.

**Comment by ldez** [comment]
Connecting to Pilot is an opt-in decision, there is just an iframe to display the UI.

**Comment by logopk** [comment]
However it is getting my request with my IP. (external system that I do not want to contact!)

**Comment by wasmitnetzen** [comment]
Why is it an Iframe then? I don't see a technical reason to use an Iframe to display a simple button except being able to count the amount of requests to that URL. i.e. tracking.

**Comment by ooshkamils515** [comment]
> Connecting to Pilot is an opt-in decision, there is just an iframe to display the Pilot UI inside the Traefik UI.
> 
> The call to display Pilot is not a tracking system.

The fact of the matter is that it is phoning home to a server that I didn't set up.  From a technical standpoint, not only can it be used as a tracking mechanism, but it is also sending enough data back to fingerprint the install via callback urls, as well as embedding information with what appears to be uptime info in the requests:

`{"Version":"2.4.7","Codename":"livarot","startDate":"2021-03-16T13:44:42.15895964-04:00"}`

Why does this "non tracking system" need this?
A lot of people use traefik in environments specifically designed to avoid cloud services. I now have to manually block requests from my own infrastructure to a cloud services that I will never use, and do not want in any way.

**Comment by ChristianCiach** [comment]
Calling something "not a tracking system" does not make something not a tracking system.

**Comment by ldez** [comment]
> Why does this "non tracking system" need this?

It's just to display the information on the page when you create your token.

**Comment by binlab** [comment]
The fact is any service that asks and collects private and personal information (IP address for example) can be called a tracking system

**Comment by ooshkamils515** [comment]
> The fact is any service that asks and collects private and personal information (IP address for example) can be called a tracking system

Exactly, and the general issue here is not that there is a iframe button, but that it can't be turned off.  Browser based preferences are not a solution - the pilot.traefik.io server would still be called at least one for every new sign in.  Ideally it should be opt-in, but at the very least there should be a config switch to turn the pilot integration/button off entirely for the entire traefik instance.

**Comment by lrsksr** [comment]
![image](https://user-images.githubusercontent.com/16703304/111653818-7b056880-8808-11eb-9fc9-aa2d82af523d.png)

No tracking, you say? I consider Google Tag Manager and Google Analytics trackers. All this is coming in through the iframe.

**Comment by ptruman** [comment]
My LAN sits behind a piHole, so Google Analytics isn't going to get
out.  I've allowed pilot, but it should be configurable.

**Comment by ldez** [comment]
I'm just honest with you: I'm just saying that we have not built that as a tracking system, you can trust me or not.

I understand your point of view, we listen to all your feedback, and we will show how could you improve that.

**Comment by bramceulemans** [comment]
It's not about if you're being honest or not, it's about the principle of opt-in for this kind "features". This kind of advertising should only show up when you configure it to do so.

**Comment by kokx** [comment]
The best (and frankly only) way to improve it, is to allow us to disable it with a flag in the config (or similar), or disable it by default.

So we do not have to rely on pinky promises that "we really do not track you".

**Comment by megnu** [comment]
I'll just be honest with you too. analytics.js = Google Analytics = tracking system

**Comment by jasonkuehl** [comment]
It might not be built that way but what's how we're taking it. It should be removed or added an on/off switch. Or we'll be honest (the community) we'll use something else.

**Comment by fhemberger** [comment]
I truly believe that this feature was not ill intended, still its implementation shows to be problematic. And the inclusion of Google Analytics in this iframe without notification of the user (i.e. data protection policy) is also a GDPR issue for all European data subjects.

**Comment by SantoDE** [comment]
Hey everyone, 

wow, this went on 🔥  pretty quick. 

Just so you know: We saw that and we started some discussions (again). Please allow me some time for a more detailed answer / action and get back to you all.

**Comment by SantoDE** [comment]
Hey everyone,

as promised, here I am :)

First of all, I am sorry for the whole situation that developed here. It was never our intention to "play you" or "spy on you" here. There is more to loose than to win for us if it were intentional. Even tough I have to agree, we did far from good here ;-)

As a consequence, we will do two actions:

* The team is already working on a PR with a boolean flag, to completely disable the loading of the IFrame if desired / wanted. 
* We are already improving he actual iframe / webapp to not track without proper consence.

We did wrong here in not making it visible enough and we are sorry for that. Thus, we are working on it now to fix it fast.

We will link the PR once it's opened so you will be aware of the progress. It will be a fix PR on the 2.4 branch, which means once merged, we will also do a release of 2.4.8 soon after to roll out the change.

For issues like that, we are also always looking for external maintainers as more eyes on critical topics like that is the better. If you are interested on that, please get in touch with me.

Also, our office hours are always a good place to discuss issues like that more directly.

Last but not least, please let me know whether you see any other improvement we can make.

**Comment by thomisus** [comment]
> Hey everyone,
> 
> as promised, here I am :)
> 
> First of all, I am sorry for the whole situation that developed here. It was never our intention to "play you" or "spy on you" here. There is more to loose than to win for us if it were intentional. Even though I have to agree, we did far from good here ;-)
> 
> As a consequence, we will do two actions:
> 
>     * The team is already working on a PR with a boolean flag, to completely disable the loading of the IFrame if desired / wanted.
> 
>     * We are already improving the actual iframe / webapp to not track without proper consent.
> 
> 
> We did wrong here in not making it visible enough and we are sorry for that. Thus, we are working on it now to fix it fast.
> 
> We will link the PR once it's opened so you will be aware of the progress. It will be a fix PR on the 2.4 branch, which means once merged, we will also do a release of 2.4.8 soon after to roll out the change.
> 
> For issues like that, we are also always looking for external maintainers as more eyes on critical topics like that is the better. If you are interested on that, please get in touch with me.
> 
> Also, our office hours are always a good place to discuss issues like that more directly.
> 
> Last but not least, please let me know whether you see any other improvement we can make.

This should be set to false by default ( i.e.: OPT-IN in accordance to European GDPR law )

**Comment by SantoDE** [comment]
As we will have changes to the platform, there will no GDPR issues left :)

**Comment by traefiker** [comment]
Closed by #7994.
